### PR TITLE
Added `Command.can_run` check to !aliases command.

### LINKED ISSUE #844: Limit aliases output to only allowed commands.
## Description
We currently have a system for aliases to be able to be output for informative reasons with the `!aliases` command. It'll show a full paginated list of all command aliases, including any commands that aren't able to be used by the invoker, meaning it's just providing unnecessary information for them that they can't really utilise. 

## Proposal
Just the same as when doing help output, check each command that has an alias if the author can run it, allowing only those they can run to be built into the command output.

### PR DESCRIPTION
**Pull request by ks129**
Closes #844

### CODE REVIEW

File: bot/cogs/alias.py [hunk 1]
```diff
@@ -34,19 +34,32 @@ def __init__(self, bot: Bot):
 
         await ctx.invoke(cmd, *args, **kwargs)
 
+    async def test_alias_can_run(self, ctx: Context, alias: str) -> bool:
```
Comment (eivl) [inline bot/cogs/alias.py:37]: There probably are some `test_` prefixes scattered around in non-test code. But I personally use the `test_` prefix for only test cases. 
My suggestion is that we change the function name to something that does not use the `test_` prefix.

File: bot/cogs/alias.py [hunk 2]
```diff
@@ -34,19 +34,32 @@ def __init__(self, bot: Bot):
...
+        """Check can user use this specific alias."""
+        # We don't want arguments, only command itself and max command parts in aliases is 2.
+        if len(alias.split()) >= 3:
+            alias = " ".join(alias.split()[0:2])
+        # Force allow `get group` due it's not actual command.
+        return alias == "get group" or await self.bot.get_command(alias).can_run(ctx)
```
Comment (eivl) [inline bot/cogs/alias.py:43]: I'm not aware of what `get group` is, this is a fault of my own, but maybe there should be a comment about this in the docstring?
I'm not saying it should be changed, it's more for a discussion than anything else.

Comment (eivl) [reply bot/cogs/alias.py:43]: also, returning both bools is pretty smart. I like it.

File: bot/cogs/alias.py [hunk 3]
```diff
@@ -34,19 +34,32 @@ def __init__(self, bot: Bot):
...
         embed = Embed(
             title='Configured aliases',
             colour=Colour.blue()
         )
+        aliases = {
+            name: value for name, value in inspect.getmembers(self)
+            if isinstance(value, Command)
+            and name.endswith("_alias")
+            and await self.test_alias_can_run(ctx, name[:-len('_alias')].replace('_', ' '))
+        }
```
Comment (eivl) [inline bot/cogs/alias.py:57]: This is really clever.

File: bot/cogs/alias.py (hunk 2 above)
Comment (ks129) [reply bot/cogs/alias.py:43]: `get group` is basically `get <group>`, where group can be `tags`, `docs`. Really `tags group` command doesn't exist, just `group` is in the alias function name. I'll add this info to docstring.

**Comment by ks129** [comment]
@eivl Addressed review.

**Comment by MarkKoz** [comment]
Closing because the cog was removed in #1159/#1198

**Pull request closed without merging**